### PR TITLE
AbstractIcon: Constant Standardization

### PR DESCRIPTION
**Pull request by Windchild292**
This moves icons to use the AbstractIcon, Camouflage, and Portrait classes for related constants instead of IPlayer and Crew. I also reordered as required as a preventative measure against null values. Further, this implements a few minor changes as part of continuing to move towards using AbstractIcon to replace our current icon system.

To review: The first three commits are the changes for this branch, the four commit is a logging and formatting rework of ScenarioLoader.

The MekHQ-side of this is required for a merge.

### CODE REVIEW

File: megamek/src/megamek/common/icons/Camouflage.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+    public Camouflage(String category, String filename, int width, int height) {
+        super(category, filename, width, height);
+    }
+    //endregion Constructors
+
+    @Override
+    public Image getBaseImage() {
+        // TODO : Implement me
+        return null;
```
Comment (SJuliez) [inline megamek/src/megamek/common/icons/Camouflage.java:47]: I suppose doing nothing is on purpose in getBaseImage. In that case the //TODO could be omitted.

**Review by SJuliez — APPROVED** [review]
Looks good to me, just two remarks/questions.

File: megamek/src/megamek/client/ui/swing/dialog/imageChooser/CamoChooser.java [hunk 2]
```diff
@@ -55,7 +56,14 @@ public CamoChooser(Window parent) {
...
     }
-    
+
+    /** Creates a dialog that allows players to choose a camo pattern. */
+    public CamoChooser(Window parent, String category, String filename) {
+        super(parent, Messages.getString("CamoChoiceDialog.select_camo_pattern"),
+                new CamoRenderer(), new CamoChooserTree());
+        showSearch(true);
+    }
+
```
[the remark this answers is not in the document]
Comment (Windchild292) [reply megamek/src/megamek/client/ui/swing/dialog/imageChooser/CamoChooser.java:66]: This is just unused for now. Piecemealing the implementation.

File: megamek/src/megamek/common/icons/Camouflage.java (hunk 1 above)
Comment (Windchild292) [reply megamek/src/megamek/common/icons/Camouflage.java:47]: This is just unused for now. Piecemealing the implementation.

**Pull request merged**